solace/src/main/java/org/apache/apex/malhar/solace/AbstractSolaceBaseInputOperator.java

```diff
@@ -0,0 +1,319 @@
+  protected transient JCSMPSession session;
+
+  protected transient Consumer consumer;
+  protected transient Consumer reliableConsumer;
+
+  protected transient int operatorId;
+  protected long windowId;
+  protected transient long lastCompletedWId;
+
+  protected transient int emitCount;
+
+  protected transient volatile boolean DRFailover = false;
+  protected transient volatile boolean TCPDisconnected = false;
+
+  protected transient BlockingQueue<BytesXMLMessage> unackedMessages; // hosts the Solace messages that need to be acked when the streaming window is OK to remove
+  protected LinkedList<Long> inFlightMessageId = new LinkedList<Long>(); //keeps track of all in flight IDs since they are not necessarily sequential
+  protected transient ArrayBlockingQueue<BytesXMLMessage> arrivedTopicMessagesToProcess;
+
+  //protected transient com.solace.dt.operator.DTSolaceOperatorInputOutput.ArrayBlockingQueue<BytesXMLMessage> arrivedMessagesToProcess;
+
+  private transient ReconnectCallbackHandler rcHandler = new ReconnectCallbackHandler();
+
+  private transient CallbackMessageHandler cbHandler = new CallbackMessageHandler();
+
+  int spinMillis;
+
+  protected transient int reconnectRetryMillis = 0;
+
+  @Override
+  public void setup(Context.OperatorContext context)
+  {
+
+    operatorId = context.getId();
+    logger.info("OperatorID from Base class: {}", operatorId);
+    spinMillis = context.getValue(com.datatorrent.api.Context.OperatorContext.SPIN_MILLIS);
+    factory = JCSMPFactory.onlyInstance();
+
+    //Required for HA and DR to try forever if set to "-1"
+    JCSMPChannelProperties channelProperties = (JCSMPChannelProperties)this.properties.getProperty(JCSMPProperties.CLIENT_CHANNEL_PROPERTIES);
+    channelProperties.setConnectRetries(Integer.parseInt(this.connectRetries));
+    channelProperties.setReconnectRetries(Integer.parseInt(this.reconnectRetries));
+
+    reconnectRetryMillis = channelProperties.getReconnectRetryWaitInMillis();
+
+
+    try {
+      session = factory.createSession(this.properties, null, new PrintingSessionEventHandler());
+    } catch (JCSMPException e) {
+      DTThrowable.rethrow(e);
+    }
+
+    //logger.debug("Properties Raw: \n{}", properties.toProperties());
+    logger.debug("Properties:\n" + properties.toString());
+    //logger.debug("\n===============================================\n");
+
+    idempotentStorageManager.setup(context);
+    lastCompletedWId = idempotentStorageManager.getLargestRecoveryWindow();
+    //logger.debug("++++++++++++++++++++Largest Completed: " + lastCompletedWId);
+
+
+  }
+
+  @Override
+  public void checkpointed(long arg0)
+  {
+    // TODO Auto-generated method stub
+
+  }
+
+  @Override
+  public void committed(long window)
+  {
+    try {
+      idempotentStorageManager.deleteUpTo(operatorId, window);
+    } catch (IOException e) {
+      DTThrowable.rethrow(e);
+    }
+
+  }
+
+  protected T processMessage(BytesXMLMessage message)
```

Comment:
This processMessage is overridden in subclasses AbstractSolaceDirectInputOperator and AbstractSolaceGuaranteedIdempotentInputOperator and those methods also call this one as super.processMessage(..). Unless this processMessage(..) is going to be used in future by other subclasses without overriding it can we call this method something else (say processMessageHelper) so it is clearer